src/std/net/json-rpc.ss

```diff
@@ -0,0 +1,301 @@
+
+(def (decode-json-rpc-response decoder request-id response-json)
+  (def (mal e)
+    (raise (malformed-response request-id: request-id response: response-json e: (error-message e))))
+  (def response (with-catch mal (cut trivial-json-object->class json-rpc-response::t response-json)))
+  (def jsonrpc (@ response jsonrpc))
+  (def result (@ response result))
+  (def error (@ response error))
+  (def id (@ response id))
+  (unless (or (void? jsonrpc) ;; a 1.0 server might fail to include this field
+              (equal? jsonrpc json-rpc-version)) ;; but a recent server must reply with same version
+    (mal "bad json_rpc_version"))
+  (unless (or (void? result) (void? error))
+    (mal "result error conflict"))
+  (unless (equal? id request-id)
+    (mal "bad id"))
+  (if (void? error)
+    (with-catch mal (cut decoder result))
+    (raise (with-catch mal (cut json->json-rpc-error error)))))
+
+;;; Server code
+
+;; http handler for json-rpc
+;; NB: This will catch any exception raised and convert it into an error notified to the client.
+;; TODO: have an optional parameter to specify a logging facility for those errors we find.
+(def (json-rpc-handler processor)
+  (lambda (req res)
+    (let/cc return
+      ;; NB: the JSON RPC over HTTP says that the client MUST specify
+      ;; application/json-rpc (preferrably) or else application/json or application/jsonrequest
+      ;; in a Content-Type header, and MUST specify and Accept header with one (or many) of them
+      ;; and that a Content-Length header MUST be present... but frankly, no one bothers,
+      ;; and enforcing any of it would make the server needlessly incompatible with clients,
+      ;; so we don't bother either.
+      (case (http-request-method req)
+        ((POST) (json-rpc-handler/POST req res processor)) ;; preferred method
+        ((GET) (json-rpc-handler/GET req res processor)) ;; mostly for testing
+        (else (http-response-write-condition res Bad-Request))))))
+
+(def (json-rpc-handler/POST req res processor)
+  (let/cc return
+    (def request-json
+      (try
+       (bytes->json (http-request-body req))
+       (catch (_)
+         (json-rpc-handler/response-json
+          res (hash ("jsonrpc" "2.0") ("error" (parser-error))))
+         (return))))
+    (json-rpc-handler/JSON res processor request-json)))
+
+(def (json-rpc-handler/GET req res processor)
+  (let/cc return
+    (def request-json
+      (try
+       (def url-params (form-url-decode (http-request-params req)))
+       (def method (assget "method" url-params (void)))
+       (def params (bytes->json
+                    (base64-string->u8vector
+                     (uri-decode
+                      (assget "params" url-params)))))
+       (def json (hash ("method" method) ("params" params)))
+       (def jsonrpc (assget "jsonrpc" url-params))
+       (when jsonrpc (hash-put! json "jsonrpc" jsonrpc))
+       (def id (assget "id" url-params))
+       (when id (hash-put! json "id" id))
```

Comment:
You could write it like this:
```
(alet (id (assget "id" url-params)
  (hash-put! json "id" id))
``` 

Same for above, makes code easier to read and follow because these are ephemeral variables.